[wasi] Enabling icu in wasi.proj breaks tests

Based on https://github.com/dotnet/runtime/pull/81052

With this patch to include the icu files in the runtime pack:
```diff
diff --git a/src/mono/wasi/wasi.proj b/src/mono/wasi/wasi.proj
index 836c68ae497..6ee8c2ccbb5 100644
--- a/src/mono/wasi/wasi.proj
+++ b/src/mono/wasi/wasi.proj
@@ -7,8 +7,8 @@

   <PropertyGroup>
     <!-- TODOWASI: icu support -->
-    <!--<ICULibDir Condition="'$(MonoWasmThreads)' != 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm', 'native', 'lib'))</ICULibDir>-->
-    <!--<ICULibDir Condition="'$(MonoWasmThreads)' == 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm-threads', 'native', 'lib'))</ICULibDir>-->
+    <ICULibDir Condition="'$(MonoWasmThreads)' != 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm', 'native', 'lib'))</ICULibDir>
+    <ICULibDir Condition="'$(MonoWasmThreads)' == 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm-threads', 'native', 'lib'))</ICULibDir>
     <WasmEnableSIMD Condition="'$(WasmEnableSIMD)' == ''">false</WasmEnableSIMD>
     <FilterSystemTimeZones Condition="'$(FilterSystemTimeZones)' == ''">false</FilterSystemTimeZones>
     <WasmObjDir>$(ArtifactsObjDir)wasm</WasmObjDir>
@@ -148,10 +148,10 @@
           DependsOnTargets="GenerateWasiPropsAndRspFiles;GenerateManagedToNative;BundleTimeZones">

     <ItemGroup>
-      <!--<ICULibNativeFiles Include="$(ICULibDir)/libicuuc.a;-->
-                                  <!--$(ICULibDir)/libicui18n.a;-->
-                                  <!--$(ICULibDir)/libicudata.a" />-->
-      <!--<ICULibFiles Include="$(ICULibDir)/*.dat" />-->
+      <ICULibNativeFiles Include="$(ICULibDir)/libicuuc.a;
+                                  $(ICULibDir)/libicui18n.a;
+                                  $(ICULibDir)/libicudata.a" />
+      <ICULibFiles Include="$(ICULibDir)/*.dat" />
     </ItemGroup>
     <PropertyGroup>
       <PInvokeTableFile>$(ArtifactsObjDir)wasm/pinvoke-table.h</PInvokeTableFile>
```

.. building a test project with:
`./dotnet.sh build /bl src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj /p:Configuration=Release /p:TargetOS=wasi /
p:TargetArchitecture=wasm /t:Test -p:WasmBuildNative=true -p:EnableAggressiveTrimming=true` fails with:

```
  Performing WASI SDK build: "/workspaces/runtime/src/mono/wasi/wasi-sdk//bin/clang" @/workspaces/runtime/artifacts/obj/System.Buffers.Tests/Release/net8.0/wasi-wasm/wasm/for-publish/clang-compile.rsp
wasm-ld : error : /workspaces/runtime/artifacts/bin/microsoft.netcore.app.runtime.wasi-wasm/Release/runtimes/wasi-wasm/native/libicudata.a: archive has no index; run ranlib to add one [/workspaces/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
wasm-ld : error : /workspaces/runtime/artifacts/bin/microsoft.netcore.app.runtime.wasi-wasm/Release/runtimes/wasi-wasm/native/libicui18n.a: archive has no index; run ranlib to add one [/workspaces/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
wasm-ld : error : /workspaces/runtime/artifacts/bin/microsoft.netcore.app.runtime.wasi-wasm/Release/runtimes/wasi-wasm/native/libicuuc.a: archive has no index; run ranlib to add one [/workspaces/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
clang-14 : error : linker command failed with exit code 1 (use -v to see invocation) [/workspaces/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
/workspaces/runtime/src/mono/wasi/build/WasiApp.Native.targets(440,5): error MSB3073: The command ""/workspaces/runtime/src/mono/wasi/wasi-sdk//bin/clang" @/workspaces/runtime/artifacts/obj/System.Buffers.Tests/Release/net8.0/wasi-wasm/wasm/for-publish/clang-compile.rsp" exited with code 1. [/workspaces/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
```

cc @pavelsavara @mkhamoyan

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -1,8 +1,8 @@
 <Dependencies>
   <ProductDependencies>
-    <Dependency Name="Microsoft.NETCore.Runtime.ICU.Transport" Version="8.0.0-preview.2.23080.3">
+    <Dependency Name="Microsoft.NETCore.Runtime.ICU.Transport" Version="8.0.0-preview.2.23103.1">
       <Uri>https://github.com/dotnet/icu</Uri>
-      <Sha>69cdceb1bf95f516673ffc93e1a68b03e361eda2</Sha>
+      <Sha>7e37e61f086c51b840e5feeb2a456d4c3601b873</Sha>
     </Dependency>
     <Dependency Name="System.Net.MsQuic.Transport" Version="7.0.0-alpha.1.22406.1">
       <Uri>https://github.com/dotnet/msquic</Uri>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -219,7 +219,7 @@
     <MicrosoftDotNetCecilVersion>0.11.4-alpha.23080.1</MicrosoftDotNetCecilVersion>
     <MicrosoftDotNetCecilPdbVersion>0.11.4-alpha.23080.1</MicrosoftDotNetCecilPdbVersion>
     <!-- ICU -->
-    <MicrosoftNETCoreRuntimeICUTransportVersion>8.0.0-preview.2.23080.3</MicrosoftNETCoreRuntimeICUTransportVersion>
+    <MicrosoftNETCoreRuntimeICUTransportVersion>8.0.0-preview.2.23103.1</MicrosoftNETCoreRuntimeICUTransportVersion>
     <!-- MsQuic -->
     <MicrosoftNativeQuicMsQuicVersion>2.1.1</MicrosoftNativeQuicMsQuicVersion>
     <SystemNetMsQuicTransportVersion>7.0.0-alpha.1.22406.1</SystemNetMsQuicTransportVersion>

--- a/src/mono/wasi/wasi.proj
+++ b/src/mono/wasi/wasi.proj
@@ -6,9 +6,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <!-- TODOWASI: icu support -->
-    <!--<ICULibDir Condition="'$(MonoWasmThreads)' != 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm', 'native', 'lib'))</ICULibDir>-->
-    <!--<ICULibDir Condition="'$(MonoWasmThreads)' == 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm-threads', 'native', 'lib'))</ICULibDir>-->
+    <ICULibDir Condition="'$(MonoWasmThreads)' != 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm', 'native', 'lib'))</ICULibDir>
+    <ICULibDir Condition="'$(MonoWasmThreads)' == 'true'">$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm-threads', 'native', 'lib'))</ICULibDir>
     <WasmEnableSIMD Condition="'$(WasmEnableSIMD)' == ''">false</WasmEnableSIMD>
     <FilterSystemTimeZones Condition="'$(FilterSystemTimeZones)' == ''">false</FilterSystemTimeZones>
     <WasmObjDir>$(ArtifactsObjDir)wasm</WasmObjDir>
@@ -149,10 +148,10 @@
           DependsOnTargets="GenerateWasiPropsAndRspFiles;GenerateManagedToNative;BundleTimeZones">
 
     <ItemGroup>
-      <!--<ICULibNativeFiles Include="$(ICULibDir)/libicuuc.a;-->
-                                  <!--$(ICULibDir)/libicui18n.a;-->
-                                  <!--$(ICULibDir)/libicudata.a" />-->
-      <!--<ICULibFiles Include="$(ICULibDir)/*.dat" />-->
+      <ICULibNativeFiles Include="$(ICULibDir)/libicuuc.a;
+                                  $(ICULibDir)/libicui18n.a;
+                                  $(ICULibDir)/libicudata.a" />
+      <ICULibFiles Include="$(ICULibDir)/*.dat" />
     </ItemGroup>
     <PropertyGroup>
       <PInvokeTableFile>$(ArtifactsObjDir)wasm/pinvoke-table.h</PInvokeTableFile>

